engine/classes/Elgg/Cache/PluginSettingsCache.php

```diff
@@ -0,0 +1,166 @@
+<?php
+namespace Elgg\Cache;
+
+use Elgg\Database;
+
+/**
+ * In memory cache of (non-user-specific, non-internal) plugin settings
+ *
+ * @access private
+ */
+class PluginSettingsCache {
+
+	/**
+	 * The cached values.
+	 *
+	 * @var array GUID => string[]
+	 */
+	protected $values = array();
```

Comment:
Can we just make everything private/final unless there is a specific point at which the class is designed for extension?